nexus/reconfigurator/planning/src/mgs_updates/mod.rs

```diff
@@ -0,0 +1,1225 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Facilities for making choices about MGS-managed updates
+
+use nexus_types::deployment::ExpectedVersion;
+use nexus_types::deployment::PendingMgsUpdate;
+use nexus_types::deployment::PendingMgsUpdateDetails;
+use nexus_types::deployment::PendingMgsUpdates;
+use nexus_types::inventory::BaseboardId;
+use nexus_types::inventory::CabooseWhich;
+use nexus_types::inventory::Collection;
+use omicron_common::api::external::TufRepoDescription;
+use slog::{debug, error, info, warn};
+use slog_error_chain::InlineErrorChain;
+use std::collections::BTreeSet;
+use std::sync::Arc;
+use thiserror::Error;
+use tufaceous_artifact::ArtifactVersion;
+use tufaceous_artifact::KnownArtifactKind;
+
+/// Generates a new set of `PendingMgsUpdates` based on:
+///
+/// * `inventory`: the latest inventory
+/// * `current_boards`: a set of baseboards to consider updating
+///   (it is possible to have baseboards in inventory that would never be
+///   updated because they're not considered part of the current system)
```

Comment:
Is this referring to the eventual full-TQ case, when we could have boards that are physically present but not trusted? Are there any other circumstances in which this would differ from what's in inventory?